app/models/stat.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class Stat
+  def initialize(name, show_in_ui: false, expose_via_api: false, &block)
+    @name = name
+    @show_in_ui = show_in_ui
+    @expose_via_api = expose_via_api
+    @block = block
+  end
+
+  attr_reader :name, :expose_via_api, :show_in_ui
+
+  def calculate
+    begin
+      return @block.call.transform_keys { |key| build_key(key) }
+    rescue StandardError => err
+      Discourse.warn_exception(
+        err,
+        message: "Unexpected error when collecting #{@name} About stats.",
+      )
+    end
+    {}
+  end
+
+  def self.all_stats
+    calculate(_all_stats)
+  end
+
+  def self.exposable_stats
+    calculate(_exposable_stats)
+  end
+
+  private
+
+  def build_key(key)
+    "#{@name}_#{key}".to_sym
+  end
+
+  def self._all_stats
```

Comment:
what is the reason of using _ ? (here and for _exposable_stats) It's already declared as private.

Reply:
That's because this class also contains public methods with names `all_stats` and `exposable_stats`.